Update API version to 2020-12-01

audunn opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
Bumps NetApp Files to api version 2020-12-01
* Added SecurityOperators to ActiveDirectory
* Volume list now gets all volumes instead of first over 100 volumes
* Added SnapshotPolicyId to NewAzNetAppFilesVolume to set snapshot policy to volume
* Added AesEncryption, LdapOverTLS, LdapSigning to ActiveDirectory
* Added FailureReason to Backup and backup patch  
* Added Encryption and systemData to NetAppAccount
* Added EncryptionKeySource to volume
<!-- Please add a brief description of the changes made in this PR -->

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

VeryEarly commented on src/NetAppFiles/NetAppFiles/Volume/GetNetAppFilesBackupStatus.cs:
```diff
@@ -0,0 +1,148 @@
+            }
+            if (existingVolume == null)
+            {
+                throw new Exception(string.Format("A Volume with name '{0}' in resource group '{1}' does not exists. Please use New-AzNetAppFilesVolume to create a new Volume.", this.Name, this.ResourceGroupName));
```
please use AzPsExceptions
https://github.com/Azure/azure-powershell-common/tree/master/src/Common/Exceptions

audunn replied:
@VeryEarly  Sure makes sense.  
Note the example code in the powershell repo should perhaps be updated to reflect this.  
https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/examples/update-cmdlet-example.md

VeryEarly commented on src/NetAppFiles/NetAppFiles/Volume/GetNetAppFilesBackupStatus.cs:
```diff
@@ -0,0 +1,148 @@
+            nameof(ResourceGroupName),
+            nameof(AccountName),
+            nameof(PoolName))]
+        public string Name { get; set; }
```
what if Name is not provided?

audunn replied:
Made mandatory

VeryEarly reviewed (changes requested):
- please update changelog
- please update psd1 for new cmdlets
- please generate help for new cmdlets
- please add/update and record test cases for new/updated cmdlets

VeryEarly commented on src/NetAppFiles/NetAppFiles/Volume/GetNetAppFilesBackupStatus.cs:
```diff
@@ -22,14 +22,15 @@
     [Cmdlet(
         "Get",
-        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NetAppFilesBackupStatus",
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NetAppFilesVolumeBackupStatus",
```
please also update filename and class name to match the cmdlet name

VeryEarly commented on src/NetAppFiles/NetAppFiles.Test/ScenarioTests/BackupTests.cs:
```diff
@@ -28,6 +28,7 @@ public BackupTests(Xunit.Abstractions.ITestOutputHelper output)
         }
 
         [Fact(Skip = "Backup service side bug causes this to fail, re-enable when fixed")]
+        //[Fact]
```
please remove line 31

VeryEarly commented on src/NetAppFiles/NetAppFiles.Test/ScenarioTests/BackupTests.cs:
```diff
@@ -40,5 +41,14 @@ public void TestBackupPipelines()
         }
+
+        [Fact(Skip = "Backup service side bug causes this to fail, re-enable when fixed")]
+        //[Fact]
```
please remove line 46

VeryEarly commented on src/NetAppFiles/NetAppFiles/ChangeLog.md:
```diff
@@ -18,6 +18,9 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Added AllowLocalNfsUsersWithLdap to ActiveDirectory
```
please specify new cmdlets in changelog

VeryEarly commented on src/NetAppFiles/NetAppFiles/Volume/GetNetAppFilesBackupStatus.cs:
```diff
@@ -0,0 +1,149 @@
+{
+    [Cmdlet(
+        "Get",
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NetAppFilesVolumeBackupStatus",
```
please update filename and class name to match cmdlet name

VeryEarly commented on src/NetAppFiles/NetAppFiles.Test/ScenarioTests/BackupTests.ps1:
```diff
@@ -292,3 +292,139 @@ function Test-BackupPipelines
+        Assert-AreEqual "$accName/$poolName/$volName1/$backupName1" $retrievedBackup.Name
+
+        # Get volume backup status
+        $retrievedBackupStatus = Get-NetAppFilesVolumeBackupStatus -ResourceGroupName $resourceLocation -AccountName $accName -PoolName $poolName -Name $volName1 
```
this name is incorrect 
should be Get-AzNetAppFilesVolumeBackupStatus

VeryEarly reviewed (changes requested):
please add "Get-AzNetAppFilesVolumeBackupStatus" to:

- NetAppFiles.psd1
- Changelog.md
- generate help markdown

audunn commented:
> please add "Get-AzNetAppFilesVolumeBackupStatus" to:
> 
> * NetAppFiles.psd1
> * Changelog.md
> * generate help markdown

@VeryEarly Get-AzNetAppFilesVolumeBackupStatus is new but due to service side issue with the operation we are thinking of not exposing it for now and expose it in next iteration. Does that work, i.e. can we leave the code here and expose it later or should it be removed?

VeryEarly commented:
> > please add "Get-AzNetAppFilesVolumeBackupStatus" to:
> > 
> > * NetAppFiles.psd1
> > * Changelog.md
> > * generate help markdown
> 
> @VeryEarly Get-AzNetAppFilesVolumeBackupStatus is new but due to service side issue with the operation we are thinking of not exposing it for now and expose it in next iteration. Does that work, i.e. can we leave the code here and expose it later or should it be removed?

I recommend to split this PR into two, move new cmdlets to another PR and target to feature release whenever service ready.

audunn commented:
> I recommend to split this PR into two, move new cmdlets to another PR and target to feature release whenever service ready.

Backup was fixed service side so keeping it in but updated and fixed some issues. Added to tests and new test runs.
@VeryEarly After pulling latest from master I am having some issues on my box with Az.Accounts when I try to update help markdown.
`Import-Module: The required module 'Az.Accounts' is not loaded. Load the module or remove the module from 'RequiredModules' in the file`
When I try to import Az.Account  I get `Import-Module: Assembly with same name is already loaded`
Az.Account was updated to 2.2.8 in master BTW.
Do you know of a fix or way around this issue?

VeryEarly commented:
/azp run
